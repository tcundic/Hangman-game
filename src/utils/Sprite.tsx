import React from "react";

const Sprite = ({
                    filename,
                    x,
                    y,
                    width,
                    height
                }: { filename: string, x: number, y: number, width: number, height: number }) => {
    if (!filename) {
        return null;
    }

    const style = {
        backgroundImage: `url(${filename})`,
        backgroundPosition: `${x * (-1)}px ${y * (-1)}px`,
        backgroundRepeat: 'no-repeat',
        width,
        height
    };

    return <div style={style} className="mb-6 mx-auto"/>;
}

export default Sprite;