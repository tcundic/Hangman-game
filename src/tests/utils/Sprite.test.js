import React from "react";
import { shallow } from "enzyme";
import Sprite from "../../utils/Sprite";

test('should render Sprite correctly', () => {
    const wrapper = shallow(<Sprite
        filename="file"
        x="1"
        y="1"
        width="1"
        height="1"
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should return null if no arguments are passed', () => {
    const wrapper = shallow(<Sprite />);
    expect(wrapper).toMatchSnapshot();
});