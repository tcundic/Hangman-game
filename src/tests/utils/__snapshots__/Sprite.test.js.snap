// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sprite correctly 1`] = `
<div
  className="mb-6 mx-auto"
  style={
    Object {
      "backgroundImage": "url(file)",
      "backgroundPosition": "-1px -1px",
      "backgroundRepeat": "no-repeat",
      "height": "1",
      "width": "1",
    }
  }
/>
`;

exports[`should return null if no arguments are passed 1`] = `""`;
